pkg/generate/app/cmd/template.go

```diff
@@ -38,9 +38,11 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 
 	// ensure the template objects are decoded
 	// TODO: in the future, this should be more automatic
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
-		err = errors.NewAggregate(errs)
-		return nil, fmt.Errorf("error processing template %s: %v", name, err)
+	if decode {
+		if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
```

Comment:
I'm not really sure what this does. However when I change `oc process` to do the decoding and then run it on [this template](https://github.com/openshift/origin/blob/master/test/testdata/template_required_params.yaml) the result is:
```yaml
apiVersion: v1
items:
- FullName: ""
  Groups: null
  Identities: null
  creationTimestamp: null
  name: somename
kind: List
metadata: {}
```
Instead of:
```yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: User
  metadata:
    name: somename
kind: List
metadata: {}
```

What am I missing?